client/web/src/insights/pages/creation/search-insight/components/search-insight-creation-content/validators.ts

```diff
@@ -12,8 +12,17 @@ export const requiredStepValueField = createRequiredValidator('Please specify a
  * Custom validator for chart series. Since series has complex type
  * we can't validate this with standard validators.
  * */
-export const seriesRequired: Validator<DataSeries[]> = series =>
-    series && series.length > 0 ? undefined : 'Series is empty. You must have at least one series for code insight.'
+export const seriesRequired: Validator<EditableDataSeries[]> = series => {
+    if (!series || series.length === 0) {
+        return 'Series is empty. You must have at least one series for code insight.'
+    }
+
+    if (series.some(series => !series.valid)) {
+        return 'Some series is invalid. Remove or edit invalid series.'
```

Comment:
If we use this I'd just add this article:
```suggestion
        return 'Some series is invalid. Remove or edit the invalid series.'
```